File: Easy/1768.merge-strings-alternately.kt
/*
 * @lc app=leetcode id=1768 lang=kotlin
 *
 * [1768] Merge Strings Alternately
 */

// @lc code=start
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        
        var string1 = word1
        var string2 = word2
        var result = ""

        while (string1.isNotEmpty() || string2.isNotEmpty())
        {
            if (string1.isNotEmpty()) {
                result += string1.first()
                string1 = string1.drop(1)
            }
            if (string2.isNotEmpty()){
                result += string2.first()
                string2 = string2.drop(1)
            }
        }
        return result
    }
}
// @lc code=end

File: Easy/1768.merge-strings-alternately.cpp
/*
 * @lc app=leetcode id=1768 lang=cpp
 *
 * [1768] Merge Strings Alternately
 */

// @lc code=start
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result;
        int x = 0, y = 0;

        while (x < word1.length() || y < word2.length()) { //either word needs to be read
            if (x < word1.length()) {
                result += word1[x]; //string[index]
                x++;
            }
            if (y < word2.length()) {
                result += word2[y];
                y++;
            }
        }
        return result;
    }
};
// @lc code=end

